packages/salesforcedx-apex-debugger/src/commands/protocol.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export interface DebuggerResponse {
```

Comment:
Was this hand generated? Or do we have a way to auto-generate this?
If we do have some fully/semi auto way to do this, could we document it somewhere in case we need to do this again?

Reply:
It was semi-automated. I documented it in an internal doc, but I can also summarize it the source.